aten/src/ATen/native/mps/operations/Linear.mm

```diff
@@ -25,6 +25,12 @@ Tensor _mps_linear(
 
   using namespace mps;
 
+  TORCH_CHECK(input.dtype() == weight.dtype(), "tensors must have the same dtype");
```

Comment:
Some of the test_mps.py tests:
```
    def test_linear2D_no_bias(self):
        self._linear_helper(in_features=2, out_features=3, shape=((4, 2)), bias=False, backward_pass=False)
```
are failing, here the bias can have UNKNOWN_SCALAR type or none dtype

```
test_linear2D_no_bias_backward (__main__.TestMPS) ... 
Input float
weight float
bias UNKNOWN_SCALAR
ERROR
```

Reply:
I think we should relax the bias type check.

Reply:
Ho the issue is that the bias can be present but undefined. Adding support for that.